packages/odata-service-inquirer/src/prompts/validators.ts

```diff
@@ -0,0 +1,81 @@
+import { readFile } from 'fs/promises';
+import { t } from '../i18n';
+import { OdataVersion } from '@sap-ux/odata-service-writer';
+import { MetadataFactory } from '@sap/wing-service-explorer';
+import LoggerHelper from './logger-helper';
+
+/**
+ * Validator function to verify if the specified metadata edmx version matches the specified required odata version.
+ *
+ * @param edmx the edmx to validate
+ * @param requiredVersion the required odata version to validate against
+ * @returns version and/or validation error message
+ */
+export function validateODataVersion(
+    edmx: string,
+    requiredVersion?: OdataVersion
+): { validationMsg?: string; version?: OdataVersion } {
+    const metadataFactory = MetadataFactory.getMetadataFactory();
+    try {
+        const explorer = metadataFactory.getMetadataExplorer(edmx);
+        // Wing service explorer does not export the type of the protocol, so we need to check the string
+        const version = explorer.getProtocolType().indexOf('v4') > 0 ? OdataVersion.v4 : OdataVersion.v2;
+
+        if (requiredVersion && requiredVersion !== version) {
+            const odataErrorMsg = t('prompts.validationMessages.odataVersionMismatch', {
+                providedOdataVersion: version,
+                requiredOdataVersion: requiredVersion
+            });
+            LoggerHelper.logger.error(odataErrorMsg);
+            return {
+                validationMsg: odataErrorMsg
+            };
+        }
+        return {
+            version
+        };
+    } catch (err) {
+        return {
+            validationMsg: t('prompts.validationMessages.metadataInvalid')
+        };
+    }
+}
+
+/**
+ * Validates the metadata file, returning an error string, if not valid, or an object with the valid metadata and version.
+ *
+ * @param path
+ * @param odataVersion
+ * @returns
+ */
+
+export const validateMetadataFile = async (
```

Comment:
The `cap-project` directory has a `validators.ts` with the associated validator for cap prompt, I'm wondernig why this isn't within the `metadat-file` dir?